docs/source/future/core/nrunner.rst

```diff
@@ -53,22 +53,100 @@ Runnable
-
-Each runnable kind may require a different amount of information to be
-provided so that it can be instantiated.  Using standalone executables
-as an example, the information required should be limtied to the
-location of the the standalone executable file.  The following
-pseudo-code may help to put these ideas together::
-
-  runnable_instance = create_runnable('exec', uri='/bin/true')
+produce some kind of result.  It's a passive entity that can not
+execute itself and can not produce results itself.
+
+This description of a runnable is abstract on purpose.  While the most
+common use case for a Runnable is to describe how to execute a
+test, there seems to be no reason to bind that concept to a
+test. Other Avocado subsystems, such as ``sysinfo``, could very well
+leverage the same concept to describe say, commands to be executed.
+
+A Runnable's ``kind``
++++++++++++++++++++++
+
+The most important information about a runnable is the declaration of
+its kind.  A kind should be a globally unique name across the entire
+Avocado community and users.
+
+When choosing a Runnable ``kind`` name, it's advisable that it should
+be:
+
+ * Informative
+ * Succinct
+ * Unique
+
+If a kind is thought to be generally useful to more than one user
+(where a user may mean a project using Avocado), it's a good idea
+to also have a generic name.  For instance, if a Runnable is going
+to describe how to run native tests for the Go programming language,
+its ``kind`` should probably be ``go``.
+
+On the other hand, if a Runnable is going to be used to describe tests
+that behave in a very peculiar way for a specific project, it's
+probably a good idea to map its ``kind`` name to the project name.
+For instance, if one is describing how to run an ``iotest`` that is
+part of the ``QEMU`` project, it may be a good idea to name this kind
+``qemu-iotest``.
+
+A Runnable's ``uri``
+++++++++++++++++++++
+
+Besides a ``kind``, each runnable kind may require a different amount
+of information to be provided so that it can be instantiated.
+
+Based on the accumulated experience so far, it's expected that a
+Runnable's ``uri`` is always going to be required.  Think of the URI
+as the one piece of information that can uniquely distinguish the
+entity (of a given ``kind``) that will be executed.
+
+If, for instance, a given runnable describes the execution of a
+executable file already present in the system, it may use its path,
+say ``/bin/true``, as its ``uri`` value.  If a runnable describes a
+web service endpoint, its ``uri`` value may just as well be its
+network URI, such as ``https://example.org:8080``.
+
+Runnable examples
++++++++++++++++++
+
+Possibly the simplest example for the use of a Runnable is to describe
+how to run a standalone executable, such as the ones available on your
+``/bin`` directory.
+
+As stated earlier, a runnable must declare its kind.  For standalone
+executables, a name such as ``exec`` fulfills the naming suggestions
+given earlier.
```

Comment:
Maybe we could clarify here the discussion that we had about "running multiple tests"  vs "trying to run the smallest test possible". What do you think?

Reply:
> Maybe we could clarify here the discussion that we had about "running multiple tests" vs "trying to run the smallest test possible". What do you think?

IMO, that kind of discussion should be located at a high-level runner explanation. I don't think this kind of explanation relates explicitly to the `Runnable examples`.